c10/macros/Macros.h

```diff
@@ -247,11 +247,19 @@ __host__ __device__
         const char* function) throw();
 }
 #endif // NDEBUG
+#if defined(__ANDROID__)
+#include <assert.h>
+#define CUDA_KERNEL_ASSERT(cond)                    \
```

Comment:
why do we need touch this macro? isn't it only used for CUDA?

Reply:
I see what you are trying to fix now. Isn't #40151 simpler fix? We already skip this for `__APPLE__` and I don't know why we need keep CUDA assert for mobile builds at all...

Reply:
I think yes, #40151 will be simpler.
Though my understanding is that CUDA_KERNEL_ASSERT is uniform assert that is used for both - cuda and CPU in the places where sources can be executed on both of them. In this case if we disable this assert for android - we will not see this assert in debug android builds. 
I do not know what is a fallback in this case - just no-op or throwing an exception?

Reply:
I did a code search and checked the original PR #31276 - this macro was mostly used in .cuh / .cu source files. The only '.h' file using it is in `c10/util/TypeCast.h` - probably this header is shared with cuda and cpu sources?

I incline towards taking simpler fix because:
1) this macro should not be used at cpu-only or mobile-only sources, as otherwise it should use TORCH_CHECK instead; between cuda build and mobile debug build, the former should have much higher usage (android debug build should be broken for a while without being noticed due to this, right?) thus a bug is less likely to first hit android debug build without being captured by some cuda or server cpu build/test.
2) the same argument can apply to `__APPLE__` and `__HIP_PLATFORM_HCC__` - we should fix them too if this is important :>

Reply:
I agree with simpler fix, I will rebase this stack once fix landed.

Reply:
And if you still think this should be fixed on Android, I'd recommend writing this `#ifdef __ANDROID__` at the same level as `__APPLE__` and `_MSC_VER` for better readability (it's already quite complicated):
```
#if defined(__APPLE__) || defined(__HIP_PLATFORM_HCC__) ...

#elif defined(__ANDROID) ...

#elif defined(_MSC_VER) ...

#else  // other platforms

#endif
```

It shouldn't share the same `NDEBUG` section in the last else branch as the `__assert_fail` will not be used.

Reply:
> I agree with simpler fix, I will rebase this stack once fix landed.

oops - I replied before seeing your comment :)

Reply:
Thanks, I  updated PR with removin changes in Macros.h